fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -1119,8 +1119,13 @@ public static void deleteStore(FDBRecordContext context, Subspace subspace) {
     public void deleteAllRecords() {
         preloadCache.invalidateAll();
         Transaction tr = ensureContextActive();
-        tr.clear(recordsSubspace().getKey(),
-                 getSubspace().range().end);
+
+        // Clear out all data except for the store header key and the index state space.
+        // Those two subspaces are determined by the configuration of the record store rather then
+        // the records.
+        Range indexStateRange = indexStateSubspace().range();
+        tr.clear(recordsSubspace().getKey(), indexStateRange.begin);
```

Comment:
Is it right to assume that e.g., `indexStateSubspace().range().begin` is always < `indexRangeSubspace().range().begin`? If not, would it be better if we just go thru the `FDBRecordStoreKeyspace` enum members and clear subspaces one by one?  I'm asking this to see how it'll affect my change.

Reply:
Hm, interesting. In some sense, that's a question about the leakiness of the abstraction, because it would be a format breaking (and therefore forbidden) change to reorder the values from that enum. I think that you can assume it's order, but it's also possible that there is a better way of doing it (and by better, I think the concern is only maintainability rather than, like, correctness or performance).